### ISSUE TITLE
Using pipe in a command for a cisco device

### ISSUE THREAD

**Issue by GitHub2022** [paramiko/paramiko#2175]
Hi, when using the pipe character to filter specific commands on a cisco device i do not get the expected result back. 
Example: show run | include 1.1.1.1

If i paste the command directly to the device the result is correct. With paramiko i only get a view lines back but not what i expect. 
Running the command without the pipe works perfectly so it is not a buffer problem. 

Escaping doesnt work at all since the backslash is send to the device. 
Using raw strings r" doesnt work as well. Same problem. 

Basically i only see the library as a problem but i can not find the root cause.

**Comment by jun66j5** [comment]
That is not a paramiko issue. Shell meta characters are unable to use because cisco device doesn't provide /bin/{sh,bash,zsh}.

**Comment by GitHub2022** [comment]
Sorry i do no get this. On the cli via ssh i can use such meta characters - why can i not send it through the channel?

**Comment by GitHub2022** [comment]
Additon. My understanding is, channel.send should send simply all character even meta to the device. From the buffer output i also see that the command has been send correctly ... somehow ... since the output is not what i expect. Doing a copy & paste from the same command and using it via SSH - it work. Can you please explain why that is?

**Comment by jun66j5** [comment]
> Doing a copy & paste from the same command and using it via SSH - it work.

Please provide what you've tried using SSH command and script which you've tried with paramiko.

**Comment by GitHub2022** [comment]
import paramiko
import logging

output_file = r'c:\users\azentsch\documents\SSH_Return.txt'

# created client using paramiko
client = paramiko.SSHClient()

# here we are loading the system host keys
client.load_system_host_keys()
client.set_missing_host_key_policy(paramiko.AutoAddPolicy())

try:
	# connecting paramiko using host
	# name and password
	client.connect(hostname, port=22, username=user, password=pw, timeout = 10)
	print("Connection done")
	return client
		
except Exception as e:
	client.close()
	print("Connection not successfull - exit")
	print(e)
	return False

if client !=False:
				
	channel = client.invoke_shell(term='vt100', width=600, height=60, width_pixels=0, height_pixels=0, environment=None)
	time.sleep(0.5)
				
	out = channel.recv(2048) 
	print (out.decode("utf-8") + "\n") 

	channel.send('term pager 0\n')
	time.sleep(0.5)

	channel.send('show run interface | include Eth \n')
	# channel.send('show run interface\n')
	time.sleep(0.5)
			
	while True:
		session_return_current_buffer = channel.recv(100000).decode("utf-8")			
		session_return_buffer_by_line = session_return_current_buffer.split('\n')
			
		for line in session_return_buffer_by_line:
			print ('Return: ' + line)
			
		print ('Buffer done with line: ')
		Print_green(session_return_buffer_by_line[-1].strip() + '\n')
			
		if '#' in session_return_buffer_by_line[-1]:
			# Check if we are back to prompt
				
			print ('*** Exit ***')
			# Terminate Loop
			break
				
		# Close Channel
		channel.close()
				
		# Close SSH
		client.close()

**Comment by GitHub2022** [comment]
found the problem - was a space character which caused the result. Pipe works like a charm.
